packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/loaders/DevLauncherAppLoaderFactory.kt

```diff
@@ -38,15 +38,16 @@ class DevLauncherAppLoaderFactory : DevLauncherKoinComponent, DevLauncherAppLoad
       // It's (maybe) a raw React Native bundle
       DevLauncherReactNativeAppLoader(url, appHost, context, controller)
     } else {
-      if (updatesInterface == null) {
+      val runtimeVersion = getRuntimeVersion(context)
+      val configuration = createUpdatesConfigurationWithUrl(url, projectUrl, runtimeVersion, installationIDHelper.getOrCreateInstallationID(context))
+
+      if (updatesInterface == null || updatesInterface?.validateUpdateWithConfiguration(configuration, context) == false) {
```

Comment:
I don't remember if kotlin allows this, but if so:

```suggestion
      if (updatesInterface == null || !updatesInterface.isValidUpdatesConfiguration(configuration, context)) {
```